src/domain/app/notificationsContext/NotificationsContext.tsx

```diff
@@ -0,0 +1,74 @@
+import uniqueId from 'lodash/uniqueId';
+import React, {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useMemo,
+  useState,
+} from 'react';
+
+import Notification, {
+  NotificationProps,
+} from '../../../common/components/notification/Notification';
+
+export type NotificationsContextProps = {
+  addNotification: (props: NotificationProps) => void;
+};
+
+export const NotificationsContext = createContext<
+  NotificationsContextProps | undefined
+>(undefined);
+
+export const NotificationsProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [notifications, setNotifications] = useState<NotificationProps[]>([]);
+
+  const value = useMemo<NotificationsContextProps>(() => {
+    return {
+      addNotification: (props: NotificationProps) =>
+        setNotifications((items) => [
+          ...items,
+          { ...props, id: uniqueId('notification-') },
+        ]),
+    };
+  }, [setNotifications]);
+
+  return (
+    <NotificationsContext.Provider value={value}>
+      {notifications.map((props, index) => {
+        const removeNotification = () =>
+          setNotifications((items) => items.filter((i) => i.id !== props.id));
```

Comment:
This line is one of the new ones not covered by tests -> maybe add a test for when a notification is closed

Reply:
I will split this PR to several smaller PRs to make it easier to review it. This part is already fixed in https://github.com/City-of-Helsinki/linkedcomponents-ui/pull/239. I will add missing PRs later today.